### PR TITLE
REF: move more of Tick into liboffsets._Tick

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/_libs/tslibs/offsets.pyx [hunk 1]
```diff
@@ -760,13 +764,78 @@ cdef class _Tick(ABCTick):
...
+                # GH#23524 if to_offset fails, we are dealing with an
+                #  incomparable type so == is False and != is True
+                other = to_offset(other)
+            except ValueError:
+                # e.g. "infer"
+                return False
+        return self.delta == other
+
+    def __ne__(self, other):
+        if isinstance(other, str):
```
Comment (jreback) [inline pandas/_libs/tslibs/offsets.pyx:816]: why is this just not self.__eq__...?

File: pandas/_libs/tslibs/offsets.pyx [hunk 2]
```diff
@@ -760,13 +764,78 @@ cdef class _Tick(ABCTick):
...
+            raise ValueError(
+                "Tick offset with `normalize=True` are not allowed."
+            )
+
+    @property
+    def delta(self) -> Timedelta:
+        return self.n * self._inc
+
+    @property
+    def nanos(self) -> int64_t:
```
Comment (jreback) [inline pandas/_libs/tslibs/offsets.pyx:792]: we should probably deprecate this (e.g. put this in _nanos for use internally), but can be later

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_libs/tslibs/offsets.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/offsets.pyx:816]: not sure, thats how it is in the status quo.  ill give it a shot and see if it breaks anything

File: pandas/_libs/tslibs/offsets.pyx (hunk 2 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/offsets.pyx:792]: maybe.  we access it in the Timedelta code, so making it private might be a minor code smell

File: pandas/_libs/tslibs/offsets.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/offsets.pyx]: doesnt break anything, just pushed with this update

**Review by jreback — APPROVED** [review]

**Pull request merged**